Support both server and client TCP sockets on all processes

is46-kim opened the pull request:
This patch helps to make a TCP connection immediately between browser and
renderer without delay. If no value in the server address parameter of
castanets, the process creates TCP server socket instead of TCP client
socket.

Desktop browser process requires 2 renderer processes.

Run 2 renderer processes as server
$ out/Default/chrome --type=renderer
Run browser process
$ out/Default/chrome --server-address=<Server Address> [URL]

Signed-off-by: Insoon Kim <is46.kim@samsung.com>

is46-kim committed:
Support both server and client TCP sockets on all processes

This patch helps to make a TCP connection immediately between browser and
renderer without delay. If no value in the server address parameter of
castanets, the process creates TCP server socket instead of TCP client
socket.

Desktop browser process requires 2 renderer processes.

Run 2 renderer processes as server
$ out/Default/chrome --type=renderer
Run browser process
$ out/Default/chrome --server-address=<Server Address> [URL]

Signed-off-by: Insoon Kim <is46.kim@samsung.com>

younghajung commented on mojo/public/cpp/platform/tcp_platform_handle_utils_posix.cc:
```diff
@@ -168,4 +158,23 @@ bool TCPServerAcceptConnection(const base::PlatformFile server_socket,
+  struct sockaddr_storage addr;
+  socklen_t len = sizeof(addr);
+  if (!getsockname(fd.get(), (struct sockaddr*)&addr, &len)) {
+    return (addr.ss_family == AF_INET);
```
udp is also AF_INET

is46-kim replied:
Currently, We don't use UDP..  The function purpose is to check if fd is AF_INET or not(UNIX domain socket).
It it OK to change the function name to IsInetSocket()?

younghajung replied:
OK, it's better to change the function name, how about IsNetworkSocket()?

is46-kim replied:
done

joonghunpark reviewed (changes requested):
Please refer to the comments in https://github.com/Samsung/Castanets/pull/266

younghajung commented on mojo/core/broker_castanets.h:
```diff
@@ -47,7 +47,7 @@ class BrokerCastanets : public Channel::Delegate, public base::SyncDelegate {
   // Returns the platform handle that should be used to establish a NodeChannel
   // to the process which is inviting us to join its network. This is the first
   // handle read off the Broker channel upon construction.
```
Please update a description as well.

is46-kim replied:
done

younghajung commented on mojo/core/broker_castanets.cc:
```diff
@@ -176,15 +153,114 @@ BrokerCastanets::BrokerCastanets(
+                         base::Unretained(this), std::move(connection_params),
+                         port, secure_connection));
+    } else
+      StartChannelOnIOThread(std::move(connection_params), port,
```
Please add curly braces here.

is46-kim replied:
done

is46-kim committed:
Move access to command line switches from mojo to content

- Remove direct access to command line switches in mojo
- Add parameters to pass tcp connection informaion to mojo

Signed-off-by: Insoon Kim <is46.kim@samsung.com>

yg48 reviewed (approved):
LGTM :)
